Revert the following commit.

This commit "fixes" one non-problem, whilst introducing a much worse problem.
It claims to fix https://github.com/pallets/jinja/issues/1253
but introduces https://github.com/pallets/jinja/issues/2069
which is much worse.

Given that 1253 only matters if you are doing something ill-advised
("setting" a variable in a template which was already passed in),
we revert this patch

commit 66587ce989e5a478e0bb165371fa2b9d42b7040f
Author: Kevin Brown-Silva <kevin@kevin-brown.com>
Date:   Mon May 2 15:33:58 2022 -0600

    Fix bug where set would sometimes fail within if
    
    There was a bug that came as the result of an early optimization done
    within ID tracking that caused loading parameters to fail in a very
    specific and rare edge case. That edge case only occurred when the
    parameter was being set within all 3 standard branches of an if block,
    since the optimization would assume that the parameter was never being
    referenced and was only ever being set. This would cause the variable to
    be set to undefined.
    
    The fix for this was to remove the optimization and still continue to
    load in the parameter even if it is set in all 3 branches.

diff --git a/src/jinja2/idtracking.py b/src/jinja2/idtracking.py
index cb4bccb0..e6dd8cd1 100644
--- a/src/jinja2/idtracking.py
+++ b/src/jinja2/idtracking.py
@@ -121,20 +121,23 @@ class Symbols:
             self._define_ref(name, load=(VAR_LOAD_RESOLVE, name))
 
     def branch_update(self, branch_symbols: t.Sequence["Symbols"]) -> None:
-        stores: t.Set[str] = set()
-
+        stores: t.Dict[str, int] = {}
         for branch in branch_symbols:
-            stores.update(branch.stores)
-
-        stores.difference_update(self.stores)
+            for target in branch.stores:
+                if target in self.stores:
+                    continue
+                stores[target] = stores.get(target, 0) + 1
 
         for sym in branch_symbols:
             self.refs.update(sym.refs)
             self.loads.update(sym.loads)
             self.stores.update(sym.stores)
 
-        for name in stores:
-            target = self.find_ref(name)
+        for name, branch_count in stores.items():
+            if branch_count == len(branch_symbols):
+                continue
+
+            target = self.find_ref(name)  # type: ignore
             assert target is not None, "should not happen"
 
             if self.parent is not None:
diff --git a/tests/test_regression.py b/tests/test_regression.py
index 10df2d1b..93d72c5e 100644
--- a/tests/test_regression.py
+++ b/tests/test_regression.py
@@ -750,16 +750,6 @@ End"""
         assert tmpl.render() == "foo"
 
 
-def test_load_parameter_when_set_in_all_if_branches(env):
-    tmpl = env.from_string(
-        "{% if True %}{{ a.b }}{% set a = 1 %}"
-        "{% elif False %}{% set a = 2 %}"
-        "{% else %}{% set a = 3 %}{% endif %}"
-        "{{ a }}"
-    )
-    assert tmpl.render(a={"b": 0}) == "01"
-
-
 @pytest.mark.parametrize("unicode_char", ["\N{FORM FEED}", "\x85"])
 def test_unicode_whitespace(env, unicode_char):
     content = "Lorem ipsum\n" + unicode_char + "\nMore text"
